### ISSUE TITLE
KeyError Handling needed for IOManager

### ISSUE THREAD

**Issue by ptomasula** [respec/HSPsquared#77]
@sjordan29's [testing](https://github.com/respec/HSPsquared/pull/73#issuecomment-1012367719) in PR #73 revealed that requesting a timeseries from the IOManager class that does not exist results in a KeyError causing the application to stop executing. I think the behavior should be to return an empty pandas.DataFrame and possibly raising a warning, but not stop execution.

**Comment by ptomasula** [comment]
I resolved the unhandled Key Error that caused model execution to stop with commit [fac5596](https://github.com/respec/HSPsquared/commit/fac5596ef58f4d54e662231fea4aeab6cb01c72c). That addressed the immediate problem related to this issue.

However this is not fully resolved. The issue was discovered when executing the test10 with the `saveall` argument set to false. In that run, the PWTGAS module requested a result timeseries that was not saved to the output file. The request timeseries was generated by a module further up in the operation sequence., but because the UCI indicates it does not need to be output, the timeseries never makes it to the IOManager for caching. 

I propose we implement some type of caching even when timeseries are not to be saved to output. I think it's probably easiest to modify the [save_timeseries](https://github.com/respec/HSPsquared/blob/e7f403aa3173c2e5ac60fca764e71d96c2379150/HSP2/utilities.py#L288) function to pass all timeseries to the IOManager. We then need to modify the [SupportsWriteTS](https://github.com/respec/HSPsquared/blob/e7f403aa3173c2e5ac60fca764e71d96c2379150/HSP2IO/protocols.py#L34rl) to accept a boolean argument indicating if the timeseries should be cached and saved, or just cached. I'll work on implementing that next week.

**Comment by ptomasula** [comment]
I've addressed this issue by modifying the `save_timeseries` function. This new approach preserves all of the columns in the dataframe instance and passes the dataframe to the `write_ts` method of `IOManager`. However, `save_timeseries` also now passes a list of the columns that should be exported. 

IOManager has also been updated to except this list of columns to export. The dataframe instance with all columns is first cached to memory, and only then are columns not in the export list selectively dropped from the dataframe and then exporting to the output file. The result is that later modules can still reference a parameter that was output from earlier modules (by accessing the cached dataframe), even if that parameter was not exported to the output file.
